config/confighttp/compression_test.go

```diff
@@ -233,6 +233,40 @@ func TestHTTPContentDecompressionHandler(t *testing.T) {
 	}
 }
 
+func TestHTTPContentCompressionNilRequestBody(t *testing.T) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+	})
+
+	addr := testutil.GetAvailableLocalAddress(t)
+	ln, err := net.Listen("tcp", addr)
+	require.NoError(t, err, "failed to create listener: %v", err)
+	srv := &http.Server{
```

Comment:
I know other tests aren't using it, but can you try to use `net/http/httptest`? It's cleaner and should be easy to use given you have it all in a handler already.